jetty-client/src/main/java/org/eclipse/jetty/client/GZIPContentDecoder.java

```diff
@@ -39,13 +47,47 @@ public GZIPContentDecoder(ByteBufferPool byteBufferPool, int bufferSize)
         super(byteBufferPool, bufferSize);
     }
 
+    @Override
+    public void beforeDecoding(HttpExchange exchange)
+    {
+        HttpFields.Mutable headers = exchange.getResponse().headers();
+        // Content-Length is not valid anymore while we are decoding.
+        headers.remove(HttpHeader.CONTENT_LENGTH);
+        // Content-Encoding should be removed/modified as the content will be decoded.
+        ListIterator<HttpField> fields = headers.listIterator();
```

Comment:
fields and headers in the same method is confusing
```suggestion
        ListIterator<HttpField> iterator = headers.listIterator();
```